File: apps/miniapp/src/lib/provider_catalog.ts
// lib/provider_catalog.ts — provider/model 启发式 helper.
//
// P6: 删静态 builtin catalog (CatalogModel / BuiltinProvider /
// BUILTIN_PROVIDERS / builtinProviderById) —— global 模型清单改读
// model-relay GET /v1/me/models (见 data/api/models.ts loadModelEntries)。
// 此文件只留:
//   - BIUMIND_OFFICIAL_MODEL: official 渠道哨兵 model id (fallback 用)
//   - providerIdForModel / modelDisplayName / providerDisplayName: 启发式

/** Official 渠道 (BiuMind 后端配的官方账号) 的哨兵 model id. */
export const BIUMIND_OFFICIAL_MODEL = 'biumind-default';

/** 反查 model id 属于哪个 provider slug (启发式: 前缀). undefined = 自定义. */
export function providerIdForModel(modelId: string): string | undefined {
  if (modelId.startsWith('claude-')) return 'anthropic';
  if (modelId.startsWith('gpt-') || modelId.startsWith('o1')) return 'openai';
  if (modelId.startsWith('gemini-')) return 'google';
  return undefined;
}

/** 给定 model id 找 displayName — P6 删 catalog 后原样返回 modelId
 *  (实际 display_name 由 /v1/me/models 提供)。 */
export function modelDisplayName(modelId: string): string {
  return modelId;
}

/** 给定 provider slug 找 displayName — 首字母大写兜底. */
export function providerDisplayName(slug: string): string {
  return slug.charAt(0).toUpperCase() + slug.slice(1);
}

File: apps/client/lib/core/llm/provider_catalog.dart
// Provider-level helpers for client-side BYOK + model dispatch.
//
// P6: 删静态 builtin catalog (BuiltinProvider / CatalogModel /
// builtinProviders / builtinProviderById) —— global 模型清单改读
// model-relay GET /v1/me/models (relay_catalog_client.dart)。此文件只留
// 两个纯函数:
//   - providerIdForModel: model id → provider slug 启发式 (无精确表)
//   - protocolForProviderSlug: standard slug → client-side 直连协议
// 之前静态 catalog 还服务 Settings UI 预置 anthropic/openai/google 行,
// 现 API Keys 页用户自填 provider, 不再需要预置行。

/// Map a model id to its provider slug (启发式: 按前缀). 返 null = 自定义 /
/// 未知 (custom provider 可用任意 id)。
String? providerIdForModel(String modelId) {
  if (modelId.startsWith('claude-')) return 'anthropic';
  // OpenAI 系列: gpt-* / o1 / o3 / o4 推理线 / chatgpt-* 别名. 仅头像图标用,
  // 路由靠 thread.providerId 不靠此 (补 o3/o4/chatgpt 防 fallback 默认图标).
  if (modelId.startsWith('gpt-') ||
      modelId.startsWith('o1') ||
      modelId.startsWith('o3') ||
      modelId.startsWith('o4') ||
      modelId.startsWith('chatgpt')) {
    return 'openai';
  }
  if (modelId.startsWith('gemini-')) return 'google';
  return null;
}

/// Map a standard provider slug → client-side direct-connect wire protocol
/// shape. Used by client_side_resolver._effectiveProtocol + the API Keys test
/// path to pick the right SSE parser. Standard providers have a fixed shape
/// (Anthropic = /v1/messages, Google = :streamGenerateContent, everything
/// else = OpenAI-compatible /v1/chat/completions). `custom` is excluded —
/// its shape is user-chosen (stored on the record) so callers must branch
/// on `provider == 'custom'` before calling this. Matches server-side
/// model-relay P1 slug-driven adaptor selection.
String protocolForProviderSlug(String slug) {
  switch (slug) {
    case 'anthropic':
      return 'anthropic';
    case 'google':
      return 'google';
    default: // openai / deepseek / doubao / dashscope / qwen / moonshot /
      //      baichuan / volcengine / azure_openai — all OpenAI-compatible.
      return 'openai_compat';
  }
}
